python/tvm/meta_schedule/cost_model/xgb_model.py

```diff
@@ -35,7 +35,24 @@
 from ..utils import cpu_count, derived_object, shash2hex
 from .metric import max_curve
 
+
+def optional_xgboost_callback(XGBoostCustomCallback):
+    # pylint:disable = import-outside-toplevel
+    try:
+        from xgboost.callback import TrainingCallback  # type: ignore
```

Comment:
Please enable `import-outside-toplevel` after this line.

Reply:
its added. Thanks for the recommendation. Integration test is running. Should be able to post the results of those in this PR once they are finished.